src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1493,12 +1493,28 @@ void PolicyHandler::OnSnapshotCreated(const BinaryMessage& pt_string) {
     return;
   }
 
-  AppIdURL app_url = policy_manager_->GetNextUpdateUrl(urls);
-  while (!IsUrlAppIdValid(app_url.first, urls)) {
-    app_url = policy_manager_->GetNextUpdateUrl(urls);
+  const uint32_t app_id = GetAppIdForSending();
+  const std::string app_policy_id =
+      application_manager_.application(app_id)->policy_app_id();
+
+  size_t app_idx = 0;
+  for (; app_idx < urls.size(); ++app_idx) {
```

Comment:
@ZhdanovP I would suggest you to replace this loop with `find_if`+ lambda
And use returned iterator instead of `urls[app_idx]` in the loop below

Reply:
@AKalinich-Luxoft Hi, if we will replace loop with find_if it will lead to more code and worse readability, moreover it requires make additional changing the signature of IsUrlAppIdValid. In my opinion it is better to use loop.